bancor_research/bancor_simulator/v3/spec/actions.py

```diff
@@ -39,10 +39,12 @@ def deposit_tkn(state: State, tkn_name: str, tkn_amt: Decimal, user_name: str) -
 
     case, bnt_increase, tkn_increase = compute_pool_depth_adjustment(state, tkn_name)
 
+    bnbnt_amt = compute_bnbnt_amt(state, bnt_increase)
     state.increase_bnt_trading_liquidity(tkn_name, bnt_increase)
     state.increase_tkn_trading_liquidity(tkn_name, tkn_increase)
     state.increase_bnt_funding_amt(tkn_name, bnt_increase)
-    state.set_protocol_wallet_balance("bnt", bnt_increase)
+    state.increase_vault_balance("bnt", bnt_increase)
+    state.increase_protocol_wallet_balance("bnt", bnbnt_amt)
 
```

Comment:
also need to add `state.increase_pooltoken_balance("bnt", bnbnt_amt)`
and `state.increase_staked_balance("bnt", bnt_increase)`

Reply:
Done